enhancements/installer/pre-flight-validations.md

```diff
@@ -0,0 +1,168 @@
+---
+title: pre-flight-validations
+authors:
+  - "@mandre"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-05-18
+last-updated: 2020-12-01
+status: implementable
+---
+
+# Pre-flight validations
+
+## Release Signoff Checklist
+
+- [v] Enhancement is `implementable`
+- [v] Design details are appropriately documented from clear requirements
+- [v] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [x] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+One of the guiding principles of OCP 4 is that the installation should always
+succeed. This goal is relatively easy to implement for public cloud platforms
+where we can make assumptions about services being available, or performance
+meeting requirements, however this is not the case with private clouds where
+each cloud is unique.
+
+It is currently not possible to tell with confidence if an installation will be
+successful or not for the following platforms:
+- BareMetal
+- OpenStack
+- oVirt
+- vSphere
+
+For this purpose, we propose to implement a framework allowing the installer to
+run pre-flight validations in order to certify that all pre-requisites are met
+to successfully install OpenShift in the selected environment.
+
+## Motivation
+
+This section is for explicitly listing the motivation, goals and non-goals of
+this proposal. Describe why the change is important and the benefits to users.
+
+### Goals
+
+By having an automated way to identify potential issues early, before the
+deployment even started, we want to reduce wasted time and resources, customer
+escalations, and improve the perception of OpenShift deployments on-premise.
+
+The installer already performs some validations:
+- checks all required fields are set and that the data is in the right format
+- some basic validation, such as networks do not overlap
```

Comment:
you can add to the list:
* Quota validations
* Flavor validations (when occurs)

But since the list isn't exhaustive, feel free to ignore my comment.

Reply:
What you say is true: we now have those. However we implemented them in the context of this very Enhancement, so I think it's fine to list them under "goals", line 66